crates/typst-syntax/src/ast.rs

```diff
@@ -1215,7 +1249,7 @@ pub enum ArrayItem<'a> {
 impl<'a> AstNode<'a> for ArrayItem<'a> {
     fn from_untyped(node: &'a SyntaxNode) -> Option<Self> {
         match node.kind() {
-            SyntaxKind::Spread => node.cast().map(Self::Spread),
+            SyntaxKind::Spread => Some(Self::Spread(Spread(node))),
```

Comment:
same comment regarding skipping `from_untyped()`.  (just adding additional markers so we know which portions of the code would be impacted by the discussion)